download-maven-plugin/src/main/java/com/googlecode/WGet.java

```diff
@@ -299,8 +315,19 @@ private void doGet(File outputFile) throws Exception {
     // TODO: this should be retrieved from wagonManager
     com.googlecode.ConsoleDownloadMonitor downloadMonitor = new com.googlecode.ConsoleDownloadMonitor();
     wagon.addTransferListener(downloadMonitor);
-    wagon.connect(repository,
-        this.wagonManager.getProxy(repository.getProtocol()));
+    if (StringUtils.isNotBlank(username)) {
+      getLog().info("providing custom authentication");
+      getLog().debug("username: " + username + " and password: ***");
+      AuthenticationInfo authInfo = new AuthenticationInfo();
+      authInfo.setUserName(username);
+      authInfo.setPassword(password);
+      wagon.connect(repository,
+              authInfo,
+              this.wagonManager.getProxy(repository.getProtocol()));
+    } else {
+      wagon.connect(repository,
```

Comment:
Would passing a dummy AuthentificationInfo (without username nor password) work here? If so, we could factorize the instantiation of AuthentificationInfo and the connect command, and only set username and password in the if block.